### PR TITLE
responds with comments for the given article id

### PR DESCRIPTION
**Pull request by liyasandeep**
completed task 9

### CODE REVIEW

File: controllers/commentsController.js [hunk 1]
```diff
@@ -0,0 +1,13 @@
+const selectCommentsByArticleId = require("../models/commentsModel.js");
+const getCommentsByArticleId = (req, res, next) => {
+  const { article_id } = req.params;
+  console.log(article_id);
```
Comment (corkesw) [inline controllers/commentsController.js]: you should remove console logs

File: models/commentsModel.js [hunk 2]
```diff
@@ -0,0 +1,16 @@
+const db = require("../db/connection.js");
+const selectCommentsByArticleId = (article_id) => {
+  const query_str = `SELECT comment_id,votes,created_at,author,body FROM comments WHERE article_id =$1 ORDER BY created_at DESC;`;
+  return db.query(query_str, [article_id]).then(({ rows: comments }) => {
+    console.log(comments);
```
Comment (corkesw) [inline models/commentsModel.js]: you should remove console logs

File: __tests__/app.test.js [hunk 3]
```diff
@@ -153,6 +153,57 @@ describe("GET/api/articles/:article_id", () => {
...
+  });
+  test("200:comments are sorted by created_at in descending order by default", () => {
+    return request(app)
+      .get("/api/articles/1/comments")
+      .expect(200)
+      .then(({ body }) => {
+        const { comments } = body;
+        expect(comments).toBeSortedBy("created_at", { descending: true });
+      });
+  });
```
Comment (corkesw) [inline __tests__/app.test.js:187]: One more happy path test needed to check an article that has no comments - in this case we would send back a 200 with an empty array

File: models/commentsModel.js [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+    if (comments.length === 0) {
```
Comment (corkesw) [inline models/commentsModel.js]: be careful here - this tells us that there are no comments but it doesn't tell us if that is because the article doesn't exist. We'll need to check this first.

**Review by corkesw — COMMENTED** [review]
Some great work here, I really like the testing and the code is very clear. The logic needs slightly more work as per the comments. Please do ass in the suggested test first before refactoring.

File: models/commentsModel.js [hunk 5]
```diff
@@ -0,0 +1,9 @@
+const db = require("../db/connection.js");
+const selectCommentsByArticleId = (article_id) => {
+  const query_str = `SELECT comment_id,votes,created_at,author,body FROM comments WHERE article_id =$1 ORDER BY created_at DESC;`;
+  return db.query(query_str, [article_id]).then(({ rows: comments }) => {
+    return comments;
```
Comment (two-triangles) [inline models/commentsModel.js:5]: You have a test for a 404 endpoint when the article ID is valid but NOT existing and I am unsure as to how this test is passing as we are not handling it inside the model. At the moment if passed an article_id of 99999, we would just return an empty array. We would also return an empty array for an article with no comments.  

We need some kind of way of checking, if we get an empty array, whether the article actually exists. Please ammend this and let me know once you have done so, so that I can have another look

Comment (liyasandeep) [reply models/commentsModel.js:5]: A the moment In line 7 of commentsController.js , i am calling   selectArticleById(article_id) and its where 404 end point gets handled as i called function to selectCommentsByArticleId() inside the then block after selectArticleById().So comments get only selected if article id is found in db.
